packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1099,6 +1126,16 @@ class ThemeData with Diagnosticable {
   ///    more aggressively than the default.
   final InteractiveInkFeatureFactory splashFactory;
 
+  /// An object containing arbitrary additions to this theme.
+  /// 
+  /// {@tool dartpad}
+  /// This sample shows how to create and use a subclass of [ThemeExtension] that
+  /// defines two colors.
+  ///
+  /// ** See code in examples/api/lib/material/theme/theme_extension.1.dart **
+  /// {@end-tool}
+  final ThemeExtension? themeExtension;
```

Comment:
nit/optional/subjective: for easier accessor maybe use function with generic:

  final ThemeExtension? _themeExtension;
  T? themeExtension<T extends ThemeExtension>() => _themeExtension is T ? T : null;

Reply:
I don't know that we want to introduce generics to `ThemeData`, WDYT @HansMuller ?

Reply:
We certainly don't want to add a type parameter to ThemeData however there may be a way to introduce the type parameter in ThemeExtension, so that FooThemeExtension is effectively a ThemeExtension<Foo>.